src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1396,10 +1396,27 @@ private uint getClassSize(CORINFO_CLASS_STRUCT_* cls)
             {
                 throw new RequiresRuntimeJitException(type);
             }
+
+            if (NeedsTypeLayoutCheck(type))
+            {
+                ISymbolNode node = _compilation.SymbolNodeFactory.CheckTypeLayout(type);
+                ((MethodWithGCInfo)_methodCodeNode).Fixups.Add(node);
+            }
 #endif
             return (uint)classSize.AsInt;
         }
 
+        bool NeedsTypeLayoutCheck(TypeDesc type)
```

Comment:
Could you move this to CorInfoImpl.ReadyToRun.cs?

IsLayoutFixedInCurrentVersionBubble lives there as well.

Reply:
Sure